### PR TITLE
Add  UNISIM-II-D reservoir model reservoir dataset

### PR DESCRIPTION
**Pull request by user27182**
Add reservoir dataset from https://github.com/pyvista/vtk-data/pull/37 to `downloads.py`.

This dataset  is also featured here:
https://examples.vtk.org/site/Python/ExplicitStructuredGrid/LoadESGrid/

The dataset is loaded as an ExplicitStructuredGrid, of which there are very few examples currently in PyVista. (No downloads are loaded as ESG, and only a basic example is featured with [load_explicit_structured](https://docs.pyvista.org/version/stable/api/examples/_autosummary/pyvista.examples.examples.load_explicit_structured))

### CODE REVIEW

**Comment by user27182** [comment]
Draft as this PR depends on https://github.com/pyvista/vtk-data/pull/37 and #5872

File: pyvista/examples/downloads.py [hunk 1]
```diff
@@ -7222,3 +7222,47 @@ def download_victorian_goblet_face_illusion(load=True):  # pragma: no cover
...
+    >>> dataset = examples.download_reservoir()
+    >>> dataset.plot()
+
+    .. seealso::
+
+        :ref:`Reservoir Dataset <reservoir_dataset>`
+            See this dataset in the Dataset Gallery for more info.
+
+    """
+    return _download_dataset(_dataset_resevoir, load=load)
```
Comment (user27182) [inline pyvista/examples/downloads.py:7261]:
```suggestion
    return _download_dataset(_dataset_reservoir, load=load)
```

File: pyvista/examples/downloads.py [hunk 2]
```diff
@@ -7222,3 +7222,47 @@ def download_victorian_goblet_face_illusion(load=True):  # pragma: no cover
...
+
+
+_dataset_resevoir = _SingleFileDownloadableLoadable(
```
Comment (user27182) [inline pyvista/examples/downloads.py:7264]:
```suggestion
_dataset_reservoir = _SingleFileDownloadableLoadable(
```

**Comment by user27182** [comment]
@pyvista-bot preview

**Comment by user27182** [comment]
@pyvista-bot preview

**Comment by user27182** [comment]
LGTM

**Comment by user27182** [comment]
@pyvista-bot preview

File: pyvista/examples/downloads.py [hunk 3]
```diff
@@ -7240,3 +7240,83 @@ def download_victorian_goblet_face_illusion(load=True):  # pragma: no cover
 _dataset_victorian_goblet_face_illusion = _SingleFileDownloadableDatasetLoader(
     'Victorian_Goblet_face_illusion/Vase.stl'
 )
+
+
+def download_reservoir(load=True):  # pragma: no cover
+    """Download the UNISIM-II-D reservoir model.
```
Comment (tkoyama010) [inline pyvista/examples/downloads.py:7246]: NITS What is the difference between UNISIM-II-D and  UNISIM-II?

Comment (user27182) [reply pyvista/examples/downloads.py:7246]: I am not sure. We could ask the original author of the data for review ? The vtu file is sourced from here https://github.com/pyvista/pyvista-support/issues/319

Comment (tkoyama010) [reply pyvista/examples/downloads.py:7294]: It is just an interest, and there is no need for that. If an outside party points this out to us, let's respond to it :)

**Review by tkoyama010 — APPROVED** [review]
LGTM
image

**Comment by user27182** [comment]
pre-commit.ci autofix

**Review by tkoyama010 — APPROVED** [review]

**Pull request merged**